src/database/utilities/utilities.js

```diff
@@ -20,3 +21,44 @@ export function addBatchToParent(batch, parent, createItem) {
   if (item.batches && item.batches.find(currentBatch => currentBatch.id === batch.id)) return;
   item.addBatch(batch);
 }
+/**
+ * Removes TransactionBatch...
+ * 1 - Remove ItemBatch from TransactionBatch
+ * 2 - Remove TransactionBatch from TransactionItem
+ * 3 - if TransactionItem has no batches, remoe TransactionItem from Transaction
+ * @param {Realm}   database   App wide database
+ * @param {Transaction}   transaction  context transaction
+ * @param {TransactionBatch} transactionBatch transaction batch to remove
+ */
+export function removeTransactionBatchUtil(database, transaction, transactionBatch) {
+  // find transaction item
+  const transactionItem =
+            transaction.items.find(testI => testI.checkTransactionBatch(transactionBatch));
+    // remove transactionBatch and it's ItemBatch
+  transactionBatch.removeItemBatch(database);
+    // removeTransactionBatch return true if transactionItem still has
+    // transactionBatches
+  if (transactionItem.removeTransactionBatch(database, transactionBatch)) {
+    database.save('TransactionItem', transactionItem);
+  } else {
+    transaction.removeTransactionItem(database, transactionItem);
+    database.save('Transaction', transaction);
+  }
+}
+/**
+ * Returns class as JSON, warning: since the schema has child <-> parent (circular))
+ * printing via console.log or json Stringify will be infinite..
+ * use util.format('[Circular]', returnJson) to log
+ * @param {Realm}   database   App wide database
+ * @param {Transaction}   transaction  context transaction
+ * @param {TransactionBatch} transactionBatch transaction batch to remove
+ */
+export function classAsJson(classInstance) {
```

Comment:
so to reply to other comment relating to Transaction.confirm, classAsJson and transactionBatch.batchAsSinglePackJson()

batchAsSinglePack was created as a non mutable function that minimises changes to .confirm .. the name kind of says it all, no real reason to go deep into it's implementation when looking at .confirm, and classAsJson might be useful in the future.. or perhaps not.  I can migrate changed to .confirm(), just thinking it might become a bit messy..
I read about option c and inlines, batchAsSinglePackJson is quite pure function, no state changes, it compartmentalises functionality and makes for easier digestion but .. good question is it worth the overhead ? even though it runs on finalise only.. 

need to talk about this in person, I am keen to change as per suggested, just want to drive my point of view a little :), I'll try to remember talking about sortDataBy